src/iuse_actor.cpp

```diff
@@ -2793,17 +2793,19 @@ repair_item_actor::repair_type repair_item_actor::default_action( const item &fi
     return RT_NOTHING;
 }
 
-bool damage_item( player &pl, item &fix )
+bool damage_item( player &pl, item_location &fix )
 {
-    pl.add_msg_if_player( m_bad, _( "You damage your %s!" ), fix.tname().c_str() );
-    if( fix.inc_damage() ) {
+    pl.add_msg_if_player( m_bad, _( "You damage your %s!" ), fix->tname() );
+    if( fix->inc_damage() ) {
         pl.add_msg_if_player( m_bad, _( "You destroy it!" ) );
-        const int pos = pl.get_item_position( &fix );
-        if( pos != INT_MIN ) {
+        if( fix.where() == item_location::type::character ) {
+            const int pos = pl.get_item_position( fix.get_item() );
             pl.i_rem_keep_contents( pos );
         } else {
-            // NOTE: Repairing items outside inventory is NOT yet supported!
-            debugmsg( "Tried to remove an item that doesn't exist" );
+            for( auto &content : fix->contents ) {
+                g->m.add_item_or_charges( fix.position(), content );
+            }
+            g->m.i_rem( fix.position(), fix.get_item() );
```

Comment:
should use `item_location::remove_item()`

Reply:
done.